src/Neo/Wallets/NEP6/NEP6Wallet.cs

```diff
@@ -345,26 +351,28 @@ private bool VerifyPasswordInternal(string password)
         public override bool ChangePassword(string oldPassword, string newPassword)
         {
             bool succeed = true;
+            NEP6Account[] accountsValues;
             lock (accounts)
             {
-                Parallel.ForEach(accounts.Values, (account, state) =>
-                {
-                    if (!account.ChangePasswordPrepare(oldPassword, newPassword))
-                    {
-                        state.Stop();
-                        succeed = false;
-                    }
-                });
+                accountsValues = accounts.Values.ToArray();
             }
+            Parallel.ForEach(accountsValues, (account, state) =>
```

Comment:
This is not thread safe if multi-`ChangePassword` requested concurrently, because  of `account.ChangePasswordPrepare` is not thread safe.

Reply:
` Parallel.ForEach` may speed up operations here, but may lead to more system overhead.

In general, it is sufficient to ensure correctness, and there is no need to over-optimize if it is not performance critical path.

> We should forget about small efficiencies, say about 97% of the time: premature optimization is the root of all evil -- Donald Knuth

Reply:
You should only use `Parallel.ForEach` if the process count is greater than `1` CPU. Otherwise you `Task.Run`. The overhead cost to much on single CPU or if the payload is small. Take in account `docker`